macOS/LocalPackages/DataBrokerProtectionShared/Sources/DataBrokerProtectionShared/VPNBypass/VPNBypassFeatureProvider.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  VPNBypassFeatureProvider.swift
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+public typealias VPNBypassFeatureProvider = VPNBypassServiceProvider & VPNConnectionStatusThroughIPCProvider
+
+public enum VPNBypassStatus: String {
+    case on
+    case off
+    case unsupported
+}
+
+public protocol VPNBypassServiceProvider: AnyObject {
+    var isSupported: Bool { get }
+    var isEnabled: Bool { get }
+    var bypassStatus: VPNBypassStatus { get }
+    var isOnboardingShown: Bool { get set }
+
+    func applyVPNBypass(_ bypass: Bool)
+}
+
+public protocol VPNConnectionStatusThroughIPCProvider {
+    func setUp()
+    var connectionStatus: String { get }
+}
+
+public extension VPNBypassServiceProvider {
+
+    var isSupported: Bool {
+#if APPSTORE
+#if NETP_SYSTEM_EXTENSION
```

Comment:
I'm not sure we can use these once we move the shared module to be outside of macOS